Sources/SCNPath/RKPathEntity.swift

```diff
@@ -0,0 +1,215 @@
+//
+//  RKPathEntity.swift
+//  SCNPath+Example
+//
+//  Created by Grant Jarvis on 3/13/21.
+//  Copyright © 2021 Max Cobb. All rights reserved.
+//
+
+import SceneKit
+import Combine
+import RealityKit
+
+public class RealityKitPathEntity : SCNPathNode {
+    
+    ///The SceneKit scene that this node belongs to.
+    public weak var scene : SCNScene?
+    
+    ///A realityKit entity which represents the generated path.
+    public var realityKitPath = Entity()
+    
+    ///The number of times we have generated a .usdz file
+    ///This is used for changing the file path each time.
+    private var entityCount = 0
+    
+    private var circleEntity : Entity?
+    
+    private var rectangleEntity : Entity?
+    
+    private var pathEntitiesInScene = [Entity]()
+    
+    override init(
+        path: [SCNVector3], width: Float = 0.5,
+        curvePoints: Float = 8, materials: [SCNMaterial] = []
+    ) {
+        super.init(path: path,
+                   width: width,
+                   curvePoints: curvePoints,
+                   materials: materials)
+        do {
+            let shapesScene = try ShapesProj.loadShapesScene()
+            loadShapes(scene: shapesScene)
+        } catch {
+            print(error.localizedDescription)
+            return
+        }
+        //Remove synchronization to save memory.
+        self.realityKitPath.visit(using: {$0.synchronization = nil})
+    }
+    
+    public required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    
+    
+    private func loadShapes(scene: ShapesProj.ShapesScene){
```

Comment:
as above, could we try this without the .reality file?

Reply:
Yes, see comment above. Now generating entirely from RealityKit for Option #2 and not loading anything from a file.